### DISCUSSION TITLE
Supported dimensions

### DISCUSSION THREAD

**Discussion by rafael2k** [fraunhoferhhi/vvenc#77] [Q&A] [answered]
Hi,
I'm writing a script to convert to any image / video to VVC. Is there any constraint for the media dimensions - like be multiple of 16 pixels or fixed supported resolutions?

**Comment by adamjw24** [comment, marked as answer]
There should be no restrictions. The encoder will internally pad the video and create appropriate signaling so that the decoder knows to only reconstruct the input dimensions.

Let us know if you find a dimensions that don't work.

**Reply by rafael2k** [reply]
Hi Adam. Here is an error:
Parameter Check Error: Error: picture height is not an integer multiple of the specified chroma subsampling
cmd line: vvencapp -i /tmp/temp-364562.yuv --qpa 1 -t 2 -r 1 -b 80000 -s 840x561 --preset medium -c yuv420 -o /tmp/temp-364562.vvc

**Reply by adamjw24** [reply]
Hi Rafael. Is this even a supported YUV420 dimension? How did you convert your video to a YUV? This would mean that the last chroma line only has one colocated luma line, while the YUV420 specifies that each chroma line is colocated with two luma lines, right?

We might need to support 444 for that. I'll put that on the agenda. I think the sources can mostly manage, but would need some testing and possibly small fix here and there.

**Reply by rafael2k** [reply]
Good question. I used imagemagick to convert a png to yuv (YCbCr 4:2:0 BT. 709). I ended up converting to resolutions always multiple of 4.
Btw, "--profile main_10" works, but "--profile main_10_stillpic" gives me argument parse error.

**Reply by adamjw24** [reply]
Its main_10_still_picture.

**Comment by rafael2k** [comment]
Thanks Adam!
